client/state/navigation/middleware.js

```diff
@@ -47,8 +47,12 @@ export const navigationMiddleware = store => {
 				return next( action );
 			case HISTORY_REPLACE:
 				if ( action.path ) {
-					// Replace the history entry, but don't dispatch a new navigation context.
-					page.replace( action.path, null, false, false );
+					if ( action.isRedirect ) {
+						page.replace( action.path );
```

Comment:
@Copons I'm not sure if I need this, but switching flows were getting stuck. For example infinite loaders for the post / not changing the view at all from classic to gutenberg.